TFUtil.py

```diff
@@ -1695,8 +1736,12 @@ def get_placeholder_flattened(self, keep_dims=False):
       dyn_axes = [(i if (i < removed_axis) else (i - 1))
                   for i in dyn_axes]
       ndim -= 1
+    shape = tf.shape(x)
+    dyn_axes.sort()
+    # Transpose remaining dyn_axes, s.t. they are together before merge
+    perm = dyn_axes + [i for i in range(self.ndim) if i not in dyn_axes]
+    x = tf.transpose(x, perm=perm)
```

Comment:
Why this change in `get_placeholder_flattened`? Please make a separate pull request, and show a test case first, where the old behavior was wrong (and why).